UCSLibraries/src/main/java/it/cnr/iit/utility/Utility.java

```diff
@@ -158,6 +138,27 @@ public static boolean createPathIfNotExists( File file ) {
         return true;
     }
 
+    /*public static String readFileAsString( String resource ) throws URISyntaxException, IOException {
+        ClassLoader classLoader = Utility.class.getClassLoader();
+        Path path = Paths.get( classLoader.getResource( resource ).toURI() );
+
+        StringBuilder sb = new StringBuilder();
+        try (BufferedReader br = new BufferedReader( new FileReader( path.toString() ) )) {
+            for( String line; ( line = br.readLine() ) != null; ) {
+                sb.append( line );
+            }
+        }
+        return sb.toString();
+    }*/
+
+    public static String readFileAsString( String resource ) throws URISyntaxException, IOException {
+        ClassLoader classLoader = Utility.class.getClassLoader();
+        Thread.interrupted(); // Avoid a nasty exception
```

Comment:
This may not be safe in production and may lead to unexpected behavior. In test, it is OK but I would avoid such a work around in production code. Lets have a look at the case where this so called "nasty exception" occurs to see if we can reorganise

Reply:
done